docs/examples/3rdparty_analysis_workflow.sh

```diff
@@ -8,94 +8,282 @@ BOBS_HOME=$(readlink -f $(mktemp -d datalad_demo.XXXX))
+# specify and later resolve data dependencies. In this example, Bob wants to
+# work with structural MRI data from the `studyforrest project`_, a public
+# brain imaging data resource. These data are made available through GitHub_,
+# so Bob can simply install the relevant dataset from this service and into
+# his own dataset.
+#
+# .. _studyforrest project: http://studyforrest.org
+# .. _github: https://github.com
+#%
+
 datalad install --source https://github.com/psychoinformatics-de/studyforrest-data-structural.git src/forrest_structural
 #datalad install --source /tmp/studyforrest-data-structural src/forrest_structural
 
-mkdir code
+#%
+# Bob has decided to collect all data inputs for his project in a subdirectory
+# ``src/``, to make it obvious which parts of his analysis steps and code
+# require 3rd-party data. Upon completion of the above command, Bob has now
+# access to the entire dataset content. However, no data was actually
+# downloaded (yet). Datalad datasets primarily contain information on a
+# dataset's content and where to obtain it, hence the installation above was
+# done rather quickly, and will still be relatively lean even for a dataset
+# that contains several hundred GBs of data.
+#
+# For his first steps Bob just needs a single file of the dataset. In order to
+# make it available locally, Bob can use the install command again, and datalad
+# will obtain it from a remote data provider.
+#%
 
-# just test data, could be
+datalad install src/forrest_structural/sub-01/anat/sub-01_T1w.nii.gz
+# just test data for now, could be
 #datalad install src/forrest_structural/sub-*/anat/sub-*_T1w.nii.gz
-echo "datalad install src/forrest_structural/sub-01/anat/sub-01_T1w.nii.gz" > code/get_required_data.sh
+
+#%
+# Although we originally installed the dataset from Github, the actual data is
+# hosted elsewhere. Datalad supports multiple redundant data providers per each
+# file in a dataset, and will transparently attempt to obtain data from an
+# alternative location if a particular data provider is not available.
+#
+# Bob wants his analysis to be easily reproducible, and therefore manages his
+# analysis scripts in the same dataset repository as the input data. Managing
+# input data, analysis code, and results the same version control system
+# creates a precise record of what version of code and input data was used to
+# create which particular results. Datalad datasets are regular Git_
+# repositories and therefore provide the same powerful source code management
+# features, as any other Git_ repository, and make them available for data too.
+#
+# .. _Git: https://git-scm.com
+#
+# Bob decided to adopt the convention to collect all of his analysis code in a
+# subdirectory ``code/`` in the root of his dataset. His first "analysis" script
+# is rather simple:
+#%
+
+mkdir code
 echo "nib-ls src/forrest_structural/sub-*/anat/sub-*_T1w.nii.gz > result.txt" > code/run_analysis.sh
```

Comment:
Side question... should we look into adding --datalad to nib-ls which would cause automatic fetching of files listed? 
Making it necessary to create a file with a list of all needed files is kinda not ideal.